cpp/src/gandiva/precompiled/time.cc

```diff
@@ -241,6 +241,44 @@ int getJanWeekOfYear(const EpochTimePoint& tp) {
   return 52;
 }
 
+static const char* WEEK[] = {"SUNDAY",   "MONDAY", "TUESDAY", "WEDNESDAY",
+                             "THURSDAY", "FRIDAY", "SATURDAY"};
+
+static const int WEEK_LEN[] = {6, 6, 7, 9, 8, 6, 8};
+
+#define NEXT_DAY_FUNC(TYPE)                                                             \
+  FORCE_INLINE                                                                          \
+  gdv_date64 next_day_from_##TYPE(gdv_int64 context, gdv_date64 millis, const char* in, \
+                                  int32_t in_len) {                                     \
+    EpochTimePoint tp(millis);                                                          \
+    const auto& day_without_hours_and_sec = tp.ClearTimeOfDay();                        \
+    const auto& presentDate = extractDow_timestamp(tp.MillisSinceEpoch());              \
+                                                                                        \
+    int dateSearch = 0;                                                                 \
+    for (int n = 0; n < 7; n++) {                                                       \
+      if (is_substr_utf8_utf8(WEEK[n], WEEK_LEN[n], in, in_len)) {                      \
+        dateSearch = n + 1;                                                             \
+        break;                                                                          \
+      }                                                                                 \
+    }                                                                                   \
+    if (dateSearch == 0) {                                                              \
+      gdv_fn_context_set_error_msg(context, "This entry not is one weekday valid");     \
+      return 0;                                                                         \
+    }                                                                                   \
+                                                                                        \
+    int32_t distanceDay = dateSearch - presentDate;                                     \
+    if (distanceDay <= 0) {                                                             \
+      distanceDay = 7 + distanceDay;                                                    \
+    }                                                                                   \
+                                                                                        \
+    gdv_date64 next_date = date_add_int64_timestamp(                                    \
+        distanceDay, day_without_hours_and_sec.MillisSinceEpoch());                     \
+                                                                                        \
+    return next_date;                                                                   \
+  }
+
+DATE_TYPES(NEXT_DAY_FUNC)
```

Comment:
You create a macro for a function that receives many types of parameters, but you only tested for timestamp